pipeline/inputs/node-exporter-metrics.md

```diff
@@ -22,6 +22,21 @@ This plugin is currently only supported on Linux based operating systems\
 | scrape_interval | The rate at which metrics are collected from the host operating system | 5 seconds |
 | path.procfs     | The mount point used to collect process information and metrics        | /proc/    |
 | path.sysfs      | The path in the filesystem used to collect system metrics              | /sys/     |
+| ne.cpu.scrape\_interval       | The rate at which cpu metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.cpufreq.scrape\_interval   | The rate at which cpufreq metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.meminfo.scrape\_interval   | The rate at which meminfo metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.diskstats.scrape\_interval | The rate at which diskstats metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.uname.scrape\_interval     | The rate at which uname metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.stat.scrape\_interval      | The rate at which stat metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.time.scrape\_interval      | The rate at which time metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.loadavg.scrape\_interval   | The rate at which loadavg metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.vmstat.scrape\_interval   | The rate at which vmstat metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| ne.filefd.scrape\_interval   | The rate at which filefd metrics are collected from the host operating system. To enable, specify greater than 0. | 0 seconds (disabled by default) |
+| metrics | To specify which metrics are collected from the host operating system | `"cpu,cpufreq,meminfo,diskstats,uname,stat,time,loadavg,vmstat,netdev,filefd"` |
```

Comment:
I believe regardless of which metrics you enable/disable you still have to mount or provide access to the same underlying filesystem components - can we add a note along these lines?

https://github.com/fluent/fluent-bit/pull/6737#issuecomment-1406211983

People may rightly assume that if I turn everything off (or whatever they don't need but as an extreme example) then they can just run an unprivileged container with no extra mounts.

Reply:
> People may rightly assume that if I turn everything off (or whatever they don't need but as an extreme example) then they can just run an unprivileged container with no extra mounts.

Sure. BTW, everything turned off should be failed to start. Everything turned off is worthless...

Reply:
Yup but it is possible - particularly if people do dynamic config via substitution or something...
That's the only way to turn on/off outputs for example dynamically using the same config - you do something like:
```
match ${OUTPUT_MATCH}
```
Then ensure the variable is set to whatever you want to match or some un-matcheable string.
I used this approach for the Couchbase sidecar for example as it is a lot simpler to just set an annotation that appears via the DownwardAPI as a variable and have the same super config everywhere - particularly for customers who do not want to learn FB. You have an image with fixed config then, just dynamically setting variables at pod deployment time.

Anyway, the point is someone could disable things but then it fails to run as we still need the things mounted even though they're not used. Let's document that at least clearly.

Reply:
Ah, dynamically configuring way should be opened up my patches. I didn't realize such use cases like an k8s deployment with environment variables.  Hmm..., I'll try to write down to clarify procfs and sysfs requirements.

Reply:
I didn't work about this comment. This will be "my homework" for the next week. :smile:

Reply:
I added procfs/sysfs requirements on https://github.com/fluent/fluent-bit-docs/pull/1034/commits/bfd3f7f0e43e9e203fe643aa157cc7376ed7a9f3.